services/ark-api/ark-api/src/ark_api/api/v1/openai.py

```diff
@@ -87,22 +87,50 @@ def process_request_metadata(
                     "error": {
                         "message": f"Invalid Ark metadata: {str(e)}",
                         "type": "invalid_request_error",
-                        "code": "invalid_ark_metadata"
+                        "code": "invalid_ark_metadata",
                     }
-                }
+                },
             )
     # Ignore other metadata keys per OpenAI SDK pattern
     return None
 
 
+# TODO: Maybe we should start streaming first, wait for the first chunk/response
+# and use the status code of that to respond with
 async def proxy_streaming_response(streaming_url: str):
     """Proxy streaming chunks from memory service."""
     timeout = httpx.Timeout(10.0, read=None)  # 10s connect, infinite read
     async with httpx.AsyncClient(timeout=timeout) as client:
         async with client.stream("GET", streaming_url) as response:
             if response.status_code != 200:
+                # Try to extract error message from response body
+                error_message = f"{response.status_code} {response.reason_phrase}"
+                try:
+                    response_text = await response.aread()
+                    response_json = json.loads(response_text.decode("utf-8"))
+                    if isinstance(response_json, dict) and "error" in response_json:
+                        if (
+                            isinstance(response_json["error"], dict)
+                            and "message" in response_json["error"]
+                        ):
+                            error_message = response_json["error"]["message"]
+                        elif isinstance(response_json["error"], str):
+                            error_message = response_json["error"]
+                except Exception:
+                    # If we can't parse the response, use the default message
+                    pass
+
+                # Forward the error response as an SSE error event
+                error_data = {
```

Comment:
nice, maybe we extract to a type perhaps same w/ incoming error? and then we have a few streaming types [perhaps useful to put them in their own file - cause we'll likely use elsewhere later]